File: leetcode/0461-hamming-distance/Solution.java
class Solution {
    public int hammingDistance(int x, int y) {
        int hammingDistance = 0;
        while (x != 0 || y != 0) {
            hammingDistance += (x & 1) ^ (y & 1);
            x = x >>> 1;
            y = y >>> 1;
        }
        
        return hammingDistance;
    }
}

File: leetcode/0461-hamming-distance/solution.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
        int xo = x ^ y;
        int ham_dist = 0;
        while (xo) {
            ham_dist += xo & 1;
            xo = xo >> 1;
        }
        return ham_dist;
    }
};